src/wipplpy/modules/connection.py

```diff
@@ -0,0 +1,43 @@
+"""
+Define the generic superclass through which to make MDSplus connections.
+
+TODO: Add a MySQL connection and maybe other connection types as well?
+"""
+
+# Standard
+import socket
+
+
+class MDSPlusConnection:
+
+    mdsplus_server = None
+    tree_name = None
+
+    def __init__(self, ...):
+        self.is_remote_connection = None
+
+    # Abstract method
+    def make_connection(self, shot_number):
+        # Choose whether to make a remote or local connection. Then return that connection.
+
+    def determine_connection_type(self, server):
+
+        local_server = socket.gethostname()
+
+        if server == local_server:
```

Comment:
There shouldn't be new lines between the `if` and `try` and elsewhere. Make it more compact.